pkg/controller/certificates/util.go

```diff
@@ -262,16 +261,17 @@ func GenerateLocallySignedTemporaryCertificate(crt *cmapi.Certificate, pkData []
 	return b, nil
 }
 
-// RenewBeforeExpiryDuration will return the amount of time before the given
-// NotAfter time that the certificate should be renewed.
-func RenewBeforeExpiryDuration(notBefore, notAfter time.Time, specRenewBefore *metav1.Duration, defaultRenewBeforeExpiryDuration time.Duration) time.Duration {
-	renewBefore := defaultRenewBeforeExpiryDuration
-	if specRenewBefore != nil {
-		renewBefore = specRenewBefore.Duration
-	}
+// CalculateRenewalTime is used to calculate when to renew a certificate. It
+// takes the user preferred renewBefore margin, from the
+// Certificate.Spec.RenewBefore field, and returns a possibly adjusted value
+// which is compatible with the duration of the actual signed certificate.
+func CalculateRenewalTime(notBefore, notAfter time.Time, renewBefore time.Duration) time.Time {
```

Comment:
Why have we changed this from being a `time.Duration` to a `time.Time`?

Reply:
Seems we could just continue to return `renewBefore` here after applying the additional 'minimum' check I think?